Flaky TestListenAndServe Server endpoints unit test on Windows

```
--- FAIL: TestListenAndServe (1.19s)
    --- FAIL: TestListenAndServe/Access_denied_to_remote_caller (0.01s)
        endpoints_windows_test.go:34: 
            	Error Trace:	D:\a\spire\spire\pkg\server\endpoints\endpoints_windows_test.go:34
            	            				D:\a\spire\spire\pkg\server\endpoints\endpoints_test.go:257
            	Error:      	Target error should be in err chain:
            	            	expected: "Access is denied."
            	            	in chain: "connection error: desc = \"transport: error while dialing: open \\\\\\\\fv-az282-803\\\\pipe\\\\spire-test-295f4c7fce7614f3: The system cannot find the file specified.\""
            	            		"open \\\\fv-az282-803\\pipe\\spire-test-295f4c7fce7614f3: The system cannot find the file specified."
            	            		"The system cannot find the file specified."
            	Test:       	TestListenAndServe/Access_denied_to_remote_caller
FAIL
FAIL	github.com/spiffe/spire/pkg/server/endpoints	1.461s
```

CI run where test failed: https://github.com/spiffe/spire/runs/7528784508?check_suite_focus=true

--- a/pkg/server/endpoints/endpoints_test.go
+++ b/pkg/server/endpoints/endpoints_test.go
@@ -253,10 +253,6 @@ func TestListenAndServe(t *testing.T) {
 	target, err := util.GetTargetName(endpoints.LocalAddr)
 	require.NoError(t, err)
 
-	t.Run("Access denied to remote caller", func(t *testing.T) {
-		testRemoteCaller(ctx, t, target)
-	})
-
 	localConn, err := util.GRPCDialContext(ctx, target, grpc.WithBlock())
 	require.NoError(t, err)
 	defer localConn.Close()
@@ -308,6 +304,10 @@ func TestListenAndServe(t *testing.T) {
 	})
 	t.Run("TrustDomain", func(t *testing.T) {
 		testTrustDomainAPI(ctx, t, localConn, noauthConn, agentConn, adminConn, downstreamConn)
+	})
+
+	t.Run("Access denied to remote caller", func(t *testing.T) {
+		testRemoteCaller(ctx, t, target)
 	})
 
 	// Assert that the bundle endpoint server was called to listen and serve

